sql/api/src/main/scala/org/apache/spark/sql/types/StringType.scala

```diff
@@ -27,7 +27,9 @@ import org.apache.spark.sql.catalyst.util.CollationFactory
  * @param collationId The id of collation for this StringType.
  */
 @Stable
-class StringType private(val collationId: Int) extends AtomicType with Serializable {
+class StringType private(val collationId: Int, var isExplicit: Boolean = false)
```

Comment:
why it's a `var`? does it need to be mutable?

Reply:
Yeah, good point, needed it to be var when it was not in the constructor. Changing it to val.